llvm/include/llvm/CASObjectFormats/Data.h

```diff
@@ -124,7 +124,8 @@ class BlockData {
   /// \p Fixups add no storage cost if they are not used.
   static void encode(uint64_t Size, uint64_t Alignment,
                      uint64_t AlignmentOffset, Optional<StringRef> Content,
-                     ArrayRef<Fixup> Fixups, SmallVectorImpl<char> &Data);
+                     ArrayRef<Fixup> Fixups, SmallVectorImpl<char> &Data,
+                     bool IsDebugInfoBlock = false);
```

Comment:
I don't think this belong here. The encode function here is simply the helper function how to encode a data block with fixups.

Debug Info Block should be a different block from block ref and the branch to select how to encode debug info block should be above this function:
```
if (IsDebugInfoBlock)
   DebugInfoBlock::encode(...)
else
   BlockData::encode(...)
```